lib/entities/scheduled-action.ts

```diff
@@ -0,0 +1,117 @@
+import { AxiosInstance } from 'axios'
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import { cloneDeep } from 'lodash'
+import {
+  DefaultElements,
+  MetaSysProps,
+  Link,
+  ISO8601Timestamp,
+  BasicCursorPaginationOptions,
+} from '../common-types'
+import { wrapCollection } from '../common-utils'
+import enhanceWithMethods from '../enhance-with-methods'
+import errorHandler from '../error-handler'
+
+/**
+ * Represents that state of the scheduled action
+ */
+enum ScheduledActionStatus {
+  /** action is pending execution */
+  scheduled = 'scheduled',
+  /** action has been started and pending completion */
+  inProgress = 'inProgress',
+  /** action was completed successfully (terminal state) */
+  succeeded = 'succeeded',
+  /** action failed to complete (terminal state) */
+  failed = 'failed',
+  /** action was canceled by a user (terminal state) */
+  canceled = 'canceled',
+}
+
+type SchedulableEntityType = 'Entry'
+type SchedulableActionType = 'publish' | 'unpublish'
+
+export type ScheduledActionSysProps = Pick<
+  MetaSysProps,
+  'id' | 'space' | 'createdAt' | 'createdBy'
```

Comment:
Some of these properties are being overwritten by the type below. Do we need to use `Pick<MetaSysProps>` at all?

Reply:
Agree, removed.